### ISSUE TITLE
afterEach has no context when using Jasmine instead of Mocha

### ISSUE THREAD

**Issue by m1gu3l** [kensho/ng-describe#57]
```
/* global ngDescribe, it, beforeEach, afterEach */
angular.module('BeforeAndAfterA', [])
  .value('foo', 'bar');

ngDescribe({
  name: 'before and after example',
  module: 'BeforeAndAfterA',
  inject: 'foo',
  only: false,
  verbose: false,
  tests: function (deps) {
    var localFoo;
	var context;

    beforeEach(function () {
      console.log('before localFoo, deps', deps);
      la(deps.foo === 'bar');
      localFoo = deps.foo;
	  la(this !== undefined, 'beforeEach has no context');
	  context = this;
    });

    it('test 1: has correct value foo', function () {
      console.log('inside test 1 localFoo, deps', deps);
      la(localFoo === 'bar');
      la(deps.foo === 'bar', 'dependencies still has foo', deps);
    });

    it('test 2: has correct value foo', function () {
      console.log('inside test 1 localFoo, deps', deps);
      la(localFoo === 'bar');
      la(deps.foo === 'bar', 'dependencies still has foo', deps);
    });
	
	it('test 3: has correct context', function () {
      la(this === context, 'test function has wrong context');
    });

    afterEach(function insideNgDescribeAfterEach() {
      console.log('after localFoo, deps', deps);
      la(localFoo === 'bar', 'localFoo is', localFoo);
      la(deps.foo === 'bar', 'dependencies still has foo', deps);
	  la(this === context, 'afterEach has wrong context');
    });
  }
});
```

**Comment by bahmutov** [comment]
I doubt this is useful to fix right now. We hacked just to get around Jasmine's wrong (in our opinion) reverse `afterEach` callback execution. Binding "this" seems like unnecessary thing. Why use `this` at all inside specs?

**Comment by m1gu3l** [comment]
Well, it was working till 1.2.0, so... 
And why use ```this```?
```
beforeEach(function() {
  this.someInstance = new deps.someComponent()
});

it('some test', function() {
  expect(this.someInstance.someProperty).toEqual(someValue);
});
```
Of course variable can be declared in scope shared by those functions, but this felt more natural.

**Comment by bahmutov** [comment]
Good point, but we had to hack around Jasmine's reverse order (issue #52) so not worth implementing something only because one testing framework has a bug. Mocha to the rescue!

**Comment by m1gu3l** [comment]
From Jasmine doc:
>The this keyword
>Another way to share variables between a beforeEach, it, and afterEach is through the this keyword. Each spec’s beforeEach/it/afterEach has the this as the same empty object that is set back to empty for the next spec’s beforeEach/it/afterEach.

```
      window.afterEach = _afterEach;

      function singleAfterEachInOrder(afterCallbacks) {
        la(check.array(afterCallbacks), 'expected array of callbacks', afterCallbacks);
        log('single "after" block with', afterCallbacks.length, 'callbacks');
        afterCallbacks.forEach(function (fn, k) {
          log('"after callback"', k, fn.name);
          fn.call(this);
        }.bind(this));
      }
      var singleCleanup = function() { singleAfterEachInOrder.call(this, afters); };
```

**Comment by m1gu3l** [comment]
It should probably be reopened and fixed with #74 and #81.

**Comment by bahmutov** [comment]
Agreed, at this point I really advise anyone to NOT use Jasmine - no releases, bugs, weird features that try to do too much - Mocha or anything really is better.
